main_test.go

```diff
@@ -72,48 +72,125 @@ func Test_filterFeatures(t *testing.T) {
 			expectedError:    nil,
 		},
 		{
-			title: "Every field which defines populated feature is populated",
+			title: "Every field which defines populated feature is populated for github",
 			planConfig: types.Plan{
-				Github: types.Github{
-					AppID:          "example ID",
-					PrivateKeyFile: "~/home/private_key.pem",
+				SCM: types.GitHubSCM,
+				TLS: true,
+				TLSConfig: types.TLSConfig{
+					DNSService: types.Route53,
 				},
+				EnableOAuth: true,
+			},
+			expectedFeatures: []string{types.DefaultFeature, types.GitHubFeature, types.Auth, types.Route53DNS},
+			expectedError:    nil,
+		},
+		{
+			title: "Every field which defines populated feature is populated for gitlab",
+			planConfig: types.Plan{
+				SCM: types.GitLabSCM,
```

Comment:
System Hook data would be good to check here

Reply:
Hey Ivana, not sure what you mean here, can you expand on this? Otherwise the objects are populated so the code can add the filters, I check only if I have the config to trigger my functions to add the filters.

Reply:
I meant a check for those secrets:
https://github.com/openfaas-incubator/ofc-bootstrap/blob/master/example.init.yaml#L79, 
but can see that we have only GitLab Instance in the type: https://github.com/openfaas-incubator/ofc-bootstrap/blob/master/pkg/types/types.go#L87

Reply:
Ah Ok I got you know, but the secret creation is not determined by that type
```
type Gitlab struct {
	GitLabInstance string `yaml:"gitlab_instance"`
}
```
, but rather
```
scm: gitlab/github
```
Before I determined by `GitHub` struct populated, but now I think the `scm` field value in the `init.yml` is way simpler

Reply:
You can check how I determine `scm_gitlab`/`scm_github` filter to be added [here](https://github.com/openfaas-incubator/ofc-bootstrap/blob/109760611a52aaed844f139e513099620a8cd71c/main.go#L662). To be more explicit `plan.SCM` not `plan.GitLab == types.GitLab{}`